dbms/src/Debug/dbgKVStore/dbgFuncInvestigator.cpp

```diff
@@ -0,0 +1,287 @@
+    result.table_name = table_name;
+
+    auto schema_syncer = tmt.getSchemaSyncerManager();
+    auto storage = tmt.getStorages().getByName(mapped_database_name, table_name, false);
+    if (storage == nullptr)
+    {
+        output(fmt::format("can't find table {} {}", mapped_database_name, table_name));
+        return;
+    }
+
+    auto table_info = storage->getTableInfo();
+    schema_syncer->syncTableSchema(context, table_info.keyspace_id, table_info.id);
+    if (table_info.partition.num > 0)
+    {
+        for (const auto & def : table_info.partition.definitions)
+        {
+            schema_syncer->syncTableSchema(context, table_info.keyspace_id, def.id);
+        }
+    }
+
+    auto table_id = table_info.id;
+
+    // tablePrefix_rowPrefix_tableID_rowID
+    TiKVKey start_key, end_key;
+    HandleID start_handle, end_handle;
+
+    constexpr static size_t OFFSET = 2;
+    if (table_info.is_common_handle)
+    {
+        size_t arg_size = args.size() - OFFSET;
+        if ((arg_size & 1) != 0)
+        {
+            throw Exception(
+                "Args not matched, should be: database-name, table-name, start1 [, start2, ..., end1, end2, ...]",
+                ErrorCodes::BAD_ARGUMENTS);
+        }
+        size_t handle_column_size = table_info.is_common_handle ? table_info.getPrimaryIndexInfo().idx_cols.size() : 1;
+        auto key_size = arg_size / 2;
+        std::vector<Field> start_field;
+        start_field.reserve(key_size);
+        std::vector<Field> end_field;
+        end_field.reserve(key_size);
+
+        for (size_t i = 0; i < handle_column_size; i++)
+        {
+            auto & column_info = table_info.columns[table_info.getPrimaryIndexInfo().idx_cols[i].offset];
+            TiDB::DatumBumpy start_datum = TiDB::DatumBumpy(
+                RegionBench::convertField(column_info, typeid_cast<const ASTLiteral &>(*args[OFFSET + i]).value),
+                column_info.tp);
+            start_field.emplace_back(start_datum.field());
+            TiDB::DatumBumpy end_datum = TiDB::DatumBumpy(
+                RegionBench::convertField(
+                    column_info,
+                    typeid_cast<const ASTLiteral &>(*args[OFFSET + key_size + i]).value),
+                column_info.tp);
+            end_field.emplace_back(end_datum.field());
+        }
+
+        start_key = RecordKVFormat::genKey(table_info, start_field);
+        end_key = RecordKVFormat::genKey(table_info, end_field);
+    }
+    else
+    {
+        start_handle = static_cast<HandleID>(safeGet<UInt64>(typeid_cast<const ASTLiteral &>(*args[OFFSET]).value));
+        start_key = RecordKVFormat::genKey(table_id, start_handle);
+        if (args.size() == 3)
+        {
+            end_handle = start_handle + 1;
+        }
+        else
+        {
+            end_handle
+                = static_cast<HandleID>(safeGet<UInt64>(typeid_cast<const ASTLiteral &>(*args[OFFSET + 1]).value));
+        }
```

Comment:
You need to add a comment about the behavior at the beginning of this function.
When I see the `end_handle` is optional, I thought it would print all the following rows after `start_handle`. But here the logic is when the `end_handle` is ignored, it will only print one row.

Reply:
fixed